Detectors/ITSMFT/ITS/base/src/SDigit.cxx

```diff
@@ -0,0 +1,236 @@
+/// \file SDigit.cxx
+/// \brief SDigit structure for ITS digits
+
+#include <TMath.h>
+#include <FairLogger.h> 
+
+#include "ITSBase/SDigit.h"
+
+using std::istream;
+using std::swap;
+
+ClassImp(AliceO2::ITS::SDigit)
+
+using namespace AliceO2::ITS;
+
+//______________________________________________________________________
+SDigit::SDigit() 
+: fChip(0)
+  ,fNTracks(0)
+  ,fROCycle(0)
+  ,fTsignal(0.0)
+  ,fNoise(0.0)
+  ,fSignalAfterElect(0.0)
+{
+  // Default constructor
+  for (int i=kBuffSize;i--;) {
+    fTrack[i] = -2;
+    fHits[i] = -1;
+    fSignal[i] = 0;
+  }
+}
+
+//______________________________________________________________________
+SDigit::SDigit(UInt_t chip,UInt_t index,Double_t noise,Int_t roCycle) 
+  :fChip(chip)
+  ,fNTracks(0)
+  ,fROCycle(roCycle)
+  ,fTsignal(0.0)
+  ,fNoise(noise)
+  ,fSignalAfterElect(0.0)
+{
+  // Standard noise constructor
+  SetUniqueID(index);
+  for (int i=kBuffSize;i--;) {
+    fTrack[i] = -2;
+    fHits[i] = -1;
+    fSignal[i] = 0;
+  }
+}
+
+//______________________________________________________________________
+SDigit::SDigit(Int_t track,Int_t hit,UInt_t chip,UInt_t index,Double_t signal,Int_t roCycle)
+  :fChip(chip)
+  ,fNTracks(1)
+  ,fROCycle(roCycle)
+  ,fTsignal(signal)
+  ,fNoise(0.0)
+  ,fSignalAfterElect(0.0)
+{
+  // Standard signal constructor
+  // Inputs:
+  //    Int_t track     The track number which produced this signal
+  //    Int_t hit       The hit number which produced this signal
+  //    Int_t chip    The chip where this signal occurred
+  //    Int_t index     The cell index where this signal occurred
+  //    Double_t signal The value of the signal (ionization)
+  //    Int_t roCycle   Read-Out cycle
+  SetUniqueID(index);
+  fTrack[0]  = track;
+  fHits[0]   = hit;
+  fSignal[0] = signal;
+  for (int i=1;i<kBuffSize;i++) {
+    fTrack[i] = -2;
+    fHits[i] = -1;
+    fSignal[i] = 0;
+  }
+}
+
+//______________________________________________________________________
+SDigit& SDigit::operator=(const SDigit &source)
+{
+  // = operator
+  if (&source!=this) {
+    this->~SDigit();
+    new(this) SDigit(source);
+  }
+  return *this;
+  //
+}
+
+//______________________________________________________________________
+SDigit::SDigit(const SDigit &source) 
+  :TObject(source)
```

Comment:
the compiler will do a good job for the copy constructor. There are no raw pointer members, so no special handling is required. I recently learned that since C++11 one can specify 
```
SomeClass(const SomeClass &) = default
```
in the class definition to tell the compiler to build a default copy constructor (and at the same time indicate that one has thought about it).